homeassistant/components/hue/v2/binary_sensor.py

```diff
@@ -84,6 +84,8 @@ class HueMotionSensor(HueBinarySensorBase):
     @property
     def is_on(self) -> bool | None:
         """Return true if the binary sensor is on."""
+        if not self.resource.enabled:
+            return False
```

Comment:
This change induces the same issue. We don't know if it is false or not. For example, with a motion sensor, there could be motion? We don't know... as the device is disabled.

Reply:
On the contrary: the device is disabled, so there can never be any detected motion ;-) That is 100% certainty, not an 'unknown'. 

It must be 'off', because, well, it _is_ switched off. And a binary that is off translates to 'Clear' for a device_class: motion.

Reply:
It is very specific for Hue where you can temporary disable the motion sensor (from controlling your lights) and its state is always False in that case. But agreed, strictly taken there can be motion while the sensor is disabled.

I'm tempted to just close this PR as its not a very big deal and Unknown is not that bad ;-)

Reply:
> On the contrary: the device is disabled, so there can never be any detected motion ;-) That is a 100% certainty, not an 'unknown'.

That is the wrong viewing angle. The point is, the device looks available from HA and reports 'no motion'. That isn't correct in any way.

Reply:
> I'm tempted to just close this PR as its not a very big deal and Unknown is not that bad ;-)

I'm tempted to actually want to get this fixed, as it is broken in the current state and not correct IMHO. Closing this PR won't fix that either.

Reply:
I agree with @frenck here that the current working is not correct.
I see 2 possibilities here:

1) Force the return of None if the sensor is disabled (displayed as Unknown in the frontend)
2) Mark the entity as unavailable if its disabled.

I have no preference

Reply:
Agreed. 1: would only be valid in current dev, 2: is preferred I think, but you raised a concern about third-party devices. So that would make 1 the middle ground.

Reply:
Adjusted the logic to force return None (unknown) when the sensor is set disabled.